openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -333,11 +333,14 @@ def get_course_cohorts(course, assignment_type=None):
         A list of CourseUserGroup objects. Empty if there are no cohorts. Does
         not check whether the course is cohorted.
     """
+    assert course or course_id, "course or course_id required"
```

Comment:
I also feel like we really want only one or the other, but not both. May be good to write as two separate assertions (so you can complain in different ways) but if you wanted to you could also `assert bool(course) ^ bool(course_id)` to cover that as well, at the expense of some readability.